align_system/algorithms/alignment_adm_component.py

```diff
@@ -126,9 +126,17 @@ def __init__(
     def run_returns(self):
         return ('chosen_choice', 'best_sample_idx')
 
-    def _midpoint_eqn(self, medical_delta, attribute_delta, total_weight=2):
+    def _midpoint_eqn(self, kdma, opt_a_value, medical_delta, attr_delta):
+        medical_weights = {"affiliation": 2, "merit": 4}
+        medical_weight = medical_weights.get(kdma, 1.0)
+
         # Midpoint equation from ADEPT
-        return 0.5 + (medical_delta - attribute_delta)/total_weight
+        if kdma == "affiliation":
+            return (opt_a_value + medical_weight*medical_delta - opt_a_value*medical_delta)/2
+        elif kdma == "merit":
+            return (opt_a_value + medical_weight*medical_delta)/4
+        else:
+            return 0.5 + (medical_weight*medical_delta - attr_delta)/2
```

Comment:
I would much rather these weights be parameterized, but since there are different formulas for some of the attributes anyway I suppose it's more understandable to these hardcoded bits together in one place.  (No changes requested here)